tests/unit/admin/test-class-wc-rest-payments-orders-controller.php

```diff
@@ -48,11 +58,13 @@ public function setUp() {
 		$this->mock_api_client       = $this->createMock( WC_Payments_API_Client::class );
 		$this->mock_gateway          = $this->createMock( WC_Payment_Gateway_WCPay::class );
 		$this->mock_customer_service = $this->createMock( WC_Payments_Customer_Service::class );
+		$this->order_service         = new WC_Payments_Order_Service();
```

Comment:
Shouldn't this be a mock (since the Order Controller is what's under test in this file)?

(Please ignore this comment if the tests aren't complete yet!)

Reply:
Some of the tests had mocks, where others used the actual service. This is due to the existing tests either checked for notes being added to the orders or they checked for the changed order status. This does mean that some tests are somewhat duplicated, but I felt it was better that way so function X can be tested that it does call another function to change order status, for example.

Reply:
I think in general we should try and mock everything apart from what we're currently testing - although I realise that  may require a re-think to the tests here, in the past they were probably doing too much and now we have better split the responsibilities we can rely on the Order service unit tests to verify that order status and notes are getting updated properly.

With that in mind, I don't think we need to address it in this PR because it's already quite large - I think maybe we could address it in a follow up PR at some point? It's definitely not high priority or blocking by any means.